docs/architecture/adr-059-test-scopes.md

```diff
@@ -0,0 +1,235 @@
+the simulation.
+
+Modules must implement [AppModuleSimulation.WeightedOperations](https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/types/module/simulation.go#L31) to define their
+simulation operations.  Note that not all modules implement this which may indicate a
+gap in current simulation test coverage.
+
+Modules not returning simulation operations: 
+
+- `auth`
+- `capability`
+- `evidence`
+- `mint`
+- `params`
+
+A separate binary, [runsim](https://github.com/cosmos/tools/tree/master/cmd/runsim), is responsible for kicking off some of these tests and
+managing their life cycle.
+
+#### Limitations
+
+- [A success](https://github.com/cosmos/cosmos-sdk/runs/7606931983?check_suite_focus=true) may take a long time to run, 7-10 minutes per simulation in CI. 
+- [Timeouts](https://github.com/cosmos/cosmos-sdk/runs/7606932295?check_suite_focus=true) sometimes occur on apparent successes without any indication why.
+- Useful error messages not provided on [failure](https://github.com/cosmos/cosmos-sdk/runs/7606932548?check_suite_focus=true) from CI, requiring a developer to run
+  the simulation locally to reproduce.
+
+### E2E tests
+
+End to end tests exercise the entire system as we understand it in as close an approximation
+to a production environment as is practical.  Presently these tests are located at
+[tests/e2e](https://github.com/cosmos/cosmos-sdk/tree/main/tests/e2e) and rely on [testutil/network](https://github.com/cosmos/cosmos-sdk/tree/main/testutil/network) to start up an in-process Tendermint node.
+
+#### Limitations
+
+In general the limitations of end to end tests are orchestration and compute cost.
+Scaffolding is required to start up and run a prod-like environment and the this
+process takes much longer to start and run than unit or integration tests.
+
+Global locks present in Tendermint code cause stateful starting/stopping to sometimes hang
+or fail intermittently when run in a CI environment.
+
+The scope of e2e tests has been complected with command line interface testing.
+
+## Decision
+
+We accept these test scopes and identify the following decisions points for each.
+
+#### Unit Tests
+
+All modules must have mocked unit test coverage.
+
+Illustrative tests should outnumber journeys in unit tests.
+
+BDD feature tests are recommended when building up illustrative and journey scenarios.
+
+Unit tests should outnumber integration tests.
+
+Unit tests must not introduce additional dependencies beyond those already present in
+production code.
+
+When module unit test introduction as per [EPIC: Unit testing of modules via mocks](https://github.com/cosmos/cosmos-sdk/issues/12398)
+results in a near complete rewrite of an integration test suite the test suite should be
+retained and moved to `/tests/integration`.  We accept the resulting test logic duplication,
+but recommend the unit tests be rewritten in BDD style.
```

Comment:
nit: if we have some tests using BDD and others not it will create an inconsistency in the repo. Could we decided BDD or not?

Personal Preference is simple go & table tests

Reply:
I believe some modules benefit from BDD tests, while others don't. So it's really on a per-module basis.

Also: we already have some BDD tests in the SDK, e.g. in ORM.